linkerd/proxy/balance/src/gauge_endpoints.rs

```diff
@@ -0,0 +1,130 @@
+use linkerd_metrics::Gauge;
+use linkerd_stack::{layer, ExtractParam, NewService, Service};
+use std::{
+    sync::Arc,
+    task::{Context, Poll},
+};
+
+/// Parameter that provides a per-balancer endpoint gauge.
+#[derive(Clone, Debug, Default)]
+pub struct EndpointsGauges {
+    pub ready: Arc<Gauge>,
+    pub pending: Arc<Gauge>,
+}
+
+/// A [`NewService`] that builds [`NewGaugeBalancerEndpoint`]s with an
```

Comment:
nit: docs check doesn't like that `NewGaugeBalancerEndpoint` isn't publicly exported
```suggestion
/// A [`NewService`] that builds `NewGaugeBalancerEndpoint`s with an
```